examples/js/exporters/GLTFExporter.js

```diff
@@ -1028,6 +1004,64 @@ THREE.GLTFExporter.prototype = {
 
 			}
 
+			var forcedIndex = false;
+
+			if ( geometry.index === null && options.forceIndices ) {
+
+				var numFaces = geometry.attributes.position.count;
+				var indices = new Uint32Array( numFaces );
+
+				for ( var i = 0; i < numFaces; i ++ ) {
+
+					indices[ i ] = i;
+
+				}
+
+				geometry.index = new THREE.Uint32BufferAttribute( indices, 1 );
+
+				forcedIndex = true;
+
+			}
+
+			var materials = Array.isArray( mesh.material ) ? mesh.material : [ mesh.material ] ;
+			var groups = Array.isArray( mesh.material ) ? mesh.geometry.groups : [ { materialIndex: 0, start: undefined, count: undefined } ];
+
+			// assuming materials.length === groups.length
+			for ( var i = 0, il = materials.length; i < il; i ++ ) {
+
+				var primitive = {
+					mode: mode,
+					attributes: attributes,
+				};
```

Comment:
Interesting interesting.... this is not how I expected primitives to be used 😅, but it looks totally valid too. I was thinking that each primitive would have different accessors, based on the start/count subslice of the mesh's attributes. And then in `GLTFLoader`, we would have to concatenate them all back together.

What you're doing seems more efficient with indexed geometry, or if you want to re-draw the entire mesh with different materials. But it also assumes the loader does _not_ do what I was planning, concatenating all of the accessors. Or if `forceIndices` had been false, and each group had different draw range, that's not handled here.

If you look at the `BrainStem.gltf` model, it has a bunch of primitives, all with different accessors. I think we should start an issue on the glTF repo to mention these different use cases and make sure they're all OK.

Reply:
Opened: https://github.com/KhronosGroup/glTF/issues/1278

Reply:
Agreed with discussing best practive in glTF thread.

I'm sure this style is very fitting to Three.js structure and requires less changes. But yea, I've missed no-indexed case... Hm...

Reply:
Added an temporal workaround so far for no-indexed geometry, force index for no-indexed geometry with multi-material. I don't think this's a good solution so just temporal.

Reply:
> But it also assumes the loader does not do what I was planning, concatenating all of the accessors.

Hm... probably I guess `GLTFLoader` ends up supporting both shared primitive accessors and different primitive accessors because the both are valid (depending on the discussion in glTF thread tho).

Reply:
`GLTFLoader` should be fine for now, since it still creates a mesh for each primitive. At least it will reuse the same Geometry in this case.

Reply:
Yup, this change is ok once I fix materials loop I commented below. The exporter exports valid glTF and I confirmed FB can load.

So my plan is we merge this change for now, and make another PR switching shared accessors to different accessors if necessary.

Reply:
Looks good! 👍